src/Compilers/Core/MSBuildTask/GenerateMSBuildEditorConfig.cs

```diff
@@ -98,7 +107,33 @@ public override bool Execute()
             }
 
             ConfigFileContents = builder.ToString();
-            return true;
+            return WriteMSBuildEditorConfig();
+        }
+
+        public bool WriteMSBuildEditorConfig()
+        {
+            try
+            {
+                if (File.Exists(FileName.ItemSpec))
```

Comment:
>FileName.ItemSpec [](http://example.com/codeflow?start=32&length=17)

Consider extracting a local variable since this is used multiple times.

Reply:
What if the target didn't specify a value for `FileName`?

Reply:
> What if the target didn't specify a value for FileName?

The `FileName` is marked with the `Required` attribute so the build will fail if it is not provided.

Incidentally, @chsienki's comment here https://github.com/dotnet/roslyn/pull/54511#discussion_r672537299 makes me think it'll be better for back-compat if we make it optional and only write if the `FileName` is explicitly provided.